staging/src/k8s.io/endpointslice/reconciler.go

```diff
@@ -552,6 +565,23 @@ func (r *reconciler) reconcileByPortMapping(
 	return slicesToCreate, slicesToUpdate, slicesToDelete, numAdded, numRemoved
 }
 
-func (r *reconciler) deleteService(namespace, name string) {
+func (r *Reconciler) DeleteService(namespace, name string) {
 	r.metricsCache.DeleteService(types.NamespacedName{Namespace: namespace, Name: name})
 }
+
+func (r *Reconciler) GetControllerName() string {
+	return r.controllerName
+}
+
+// ManagedByChanged returns true if one of the provided EndpointSlices is
+// managed by the EndpointSlice controller while the other is not.
+func (r *Reconciler) ManagedByChanged(endpointSlice1, endpointSlice2 *discovery.EndpointSlice) bool {
+	return r.ManagedByController(endpointSlice1) != r.ManagedByController(endpointSlice2)
+}
+
+// ManagedByController returns true if the controller of the provided
+// EndpointSlices is the EndpointSlice controller.
+func (r *Reconciler) ManagedByController(endpointSlice *discovery.EndpointSlice) bool {
```

Comment:
Same as above. Can be moved to `pkg/controller/endpointslice/endpointslice_controller.go` and made private.

Reply:
same comment as above, you want to have that function as library, to identify if the particular object is managed by this controller.

Reply:
agree with Maciej